### ISSUE TITLE
Lossless transcoding of lossless, arithmetic, hdr legacy JPEG

### ISSUE THREAD

**Issue by gitoss** [libjxl/libjxl#2685]
**Describe the bug**
The official JPEG-XL website states "Lossless JPEG transcoding: about 20% smaller" - however, this some perfectly in-spec jpeg encoding methods are not supported and result only in a cryptic error message "JxlEncoderAddJPEGFrame() failed".

**To Reproduce**
Encode 1. lossless jpeg (cjpeg -lossless 7), 2. arithmetic jpeg (cjpeg -arithmetic), 3. hdr jpeg (cjpeg -precision 12) and convert with cjxl --lossless_jpeg=1. I've attached respective samples.

**Expected behavior**
Expected behavior would either be a verbose error message (lossless | arithmetic | hdr not supported) or ideally a lossless conversion to jxl. The arithmetic encoding is easy to remove tough (jpegtran -optimize).

**Environment**
 - OS: Windows
 - Compiler version: https://artifacts.lucaversari.it/libjxl/libjxl/latest/jxl-x64-windows-static.zip
 - CPU type: x64
 - cjxl/djxl version string: cjxl v0.9.0 9b275e5 [SSE4,SSSE3,SSE2]

**Additional context**
![sample-arithmetic](https://github.com/libjxl/libjxl/assets/10775306/21223d30-2c6d-47c6-9185-495a984ca79b)
![sample-hdr12](https://github.com/libjxl/libjxl/assets/10775306/7905df08-0fc2-400e-a64b-66f3221bce0b)
![sample-lossless](https://github.com/libjxl/libjxl/assets/10775306/7e4980f2-58c8-4f71-8816-c78a98207bc4)

**Comment by gitoss** [comment]
Another case seem to be cmyk legacy jpeg files: https://github.com/libjxl/libjxl/issues/2542

**Comment by gitoss** [comment]
Right, I didn't see https://github.com/libjxl/libjxl/issues/1172 - that doesn't mention 12-bit jpeg though :-)

I'm leaving the ticket open until a master "better error messages" ticket references it, or it's resolved with an at least slightly less cryptic error message in the code.

**Comment by wwinniww** [comment]
Mentioning HDR JPEG,
are there plans to support lossless conversation of files in Ultra HDR Image format from Google to JXL?

[https://developer.android.com/guide/topics/media/platform/hdr-image-format](url)
